jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -110,16 +110,22 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         }
         
         baseRequest.setHandled(true);
-        response.setContentType(MimeTypes.Type.TEXT_HTML_8859_1.asString());    
-        if (_cacheControl!=null)
-            response.setHeader(HttpHeader.CACHE_CONTROL.asString(), _cacheControl);
-        ByteArrayISO8859Writer writer= new ByteArrayISO8859Writer(4096);
-        String reason=(response instanceof Response)?((Response)response).getReason():null;
-        handleErrorPage(request, writer, response.getStatus(), reason);
-        writer.flush();
-        response.setContentLength(writer.size());
-        writer.writeTo(response.getOutputStream());
-        writer.destroy();
+
+        // Issue #124 - Don't produce text/html if the request doesn't accept it
+        String reqAccept = request.getHeader("Accept");
+        if (reqAccept == null || reqAccept.contains("text/html") || reqAccept.contains("*/*"))
```

Comment:
@joakime it is the HttpField.contains method that does the q processing, so use 
```` java
HttpField accept=getHttpFields().getField(HttpHeader.ACCEPT); 
if (accept==null || accept.contains("text/html") || accept.contains("*/*))
````